### PR TITLE
Moving terraform modules from gotc-deploy to this repo

### PR DESCRIPTION
**Pull request by hjfbynara**
added cloud-nat module
added firewall rule module
added ssl-cert module
added global ip module

I suspect that there will a follow up PRs that result from converting the configurations that were using the private TF modules to these.  But I would like to get these in place so it will be easier to iterate.

### CODE REVIEW

File: terraform-modules/global-ip-dns/test/test.tf [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+module "test_auth_proxy" {
+  source              = "../"
+
+  providers = { 
+    google = google
+    google.dns_provider = google
+  }
+  auth_proxy_dns_name = "test-auth-proxy"
+  auth_proxy_dns_zone = "gotc-dev"
```
Comment (Fab-T) [inline terraform-modules/global-ip-dns/test/test.tf:13]: I believe test should be updated later on when we are all set with variables.

File: terraform-modules/global-ip-dns/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+`google`: used for all resources 
+
+
+## Inputs
+| Name | Description | Type | Default | Required |
+|------|-------------|:----:|:-----:|:-----:|
+| auth_proxy_dns_name | DNS name used for A record | string | auth-proxy | no |
+| auth_proxy_dns_zone | DNS zone resource name | string | null | no |
+| ssl_certificate_cert | certificate for ssl cert | string | null | yes |
```
Comment (Fab-T) [inline terraform-modules/global-ip-dns/README.md:29]: do we require ssl cert for a dns entry ?

Comment (hjfbynara) [reply terraform-modules/global-ip-dns/README.md]: oops - removed that with latest commit

File: terraform-modules/global-ip-dns/test/test.tf (hunk 1 above)
Comment (hjfbynara) [reply terraform-modules/global-ip-dns/test/test.tf:13]: yep

**Pull request merged**